### ISSUE TITLE
Can't use multiple domains with entity roles or groups

### ISSUE THREAD

**Issue by melindaloubser1** [RasaHQ/rasa#10766]
### Rasa Open Source version

2.8.22 - also happens with 3.0.5

### Python version

3.8

### What operating system are you using?

OSX

### What happened?
You can reproduce by cloning https://github.com/melindaloubser1/moodbot/tree/entity_roles_multidomain and running `rasa train --domain domains`

I have a split domain setup like this:
```
├── domains
│   ├── domain.yml
│   ├── entities.yml
│   └── intents.yml
```

In entities.yml, I have an entity with roles:
```
version: '2.0'
entities:
   - city:
       roles:
       - from
       - to
```
 When I try to train a model with these domains, it fails with the error below i.e. it cannot combine the sections because it is expecting a list, but roles are a dictionary. 
 The same error does not occur when I have no roles for my entity.

### Command / Request

```shell
rasa train --domain domains
```

### Relevant log output

```shell
Traceback (most recent call last):
  File "/Users/melinda/.pyenv/versions/rasa2/bin/rasa", line 8, in <module>
    sys.exit(main())
  File "/Users/melinda/.pyenv/versions/rasa2/lib/python3.8/site-packages/rasa/__main__.py", line 118, in main
    cmdline_arguments.func(cmdline_arguments)
  File "/Users/melinda/.pyenv/versions/rasa2/lib/python3.8/site-packages/rasa/cli/train.py", line 59, in <lambda>
    train_parser.set_defaults(func=lambda args: run_training(args, can_exit=True))
  File "/Users/melinda/.pyenv/versions/rasa2/lib/python3.8/site-packages/rasa/cli/train.py", line 91, in run_training
    training_result = train_all(
  File "/Users/melinda/.pyenv/versions/rasa2/lib/python3.8/site-packages/rasa/api.py", line 109, in train
    return rasa.utils.common.run_in_loop(
  File "/Users/melinda/.pyenv/versions/rasa2/lib/python3.8/site-packages/rasa/utils/common.py", line 296, in run_in_loop
    result = loop.run_until_complete(f)
  File "uvloop/loop.pyx", line 1456, in uvloop.loop.Loop.run_until_complete
  File "/Users/melinda/.pyenv/versions/rasa2/lib/python3.8/site-packages/rasa/model_training.py", line 100, in train_async
    domain = await file_importer.get_domain()
  File "/Users/melinda/.pyenv/versions/rasa2/lib/python3.8/site-packages/rasa/shared/utils/common.py", line 141, in decorated
    return await cache.cached_result()
  File "/Users/melinda/.pyenv/versions/rasa2/lib/python3.8/site-packages/rasa/shared/importers/importer.py", line 448, in get_domain
    original, e2e_domain = await asyncio.gather(
  File "/Users/melinda/.pyenv/versions/rasa2/lib/python3.8/site-packages/rasa/shared/importers/importer.py", line 455, in _get_domain_with_e2e_actions
    stories = await self.get_stories()
  File "/Users/melinda/.pyenv/versions/rasa2/lib/python3.8/site-packages/rasa/shared/importers/importer.py", line 490, in get_stories
    return await self.importer.get_stories(
  File "/Users/melinda/.pyenv/versions/rasa2/lib/python3.8/site-packages/rasa/shared/importers/importer.py", line 399, in get_stories
    return await self._importer.get_stories(
  File "/Users/melinda/.pyenv/versions/rasa2/lib/python3.8/site-packages/rasa/shared/utils/common.py", line 141, in decorated
    return await cache.cached_result()
  File "/Users/melinda/.pyenv/versions/rasa2/lib/python3.8/site-packages/rasa/shared/importers/importer.py", line 272, in get_stories
    stories = await asyncio.gather(*stories)
  File "/Users/melinda/.pyenv/versions/rasa2/lib/python3.8/site-packages/rasa/shared/importers/rasa.py", line 55, in get_stories
    await self.get_domain(),
  File "/Users/melinda/.pyenv/versions/rasa2/lib/python3.8/site-packages/rasa/shared/importers/rasa.py", line 79, in get_domain
    domain = Domain.load(self._domain_path)
  File "/Users/melinda/.pyenv/versions/rasa2/lib/python3.8/site-packages/rasa/shared/core/domain.py", line 143, in load
    other = cls.from_path(path)
  File "/Users/melinda/.pyenv/versions/rasa2/lib/python3.8/site-packages/rasa/shared/core/domain.py", line 155, in from_path
    domain = cls.from_directory(path)
  File "/Users/melinda/.pyenv/versions/rasa2/lib/python3.8/site-packages/rasa/shared/core/domain.py", line 239, in from_directory
    domain = other.merge(domain)
  File "/Users/melinda/.pyenv/versions/rasa2/lib/python3.8/site-packages/rasa/shared/core/domain.py", line 305, in merge
    combined[key] = merge_lists(combined[key], domain_dict[key])
  File "/Users/melinda/.pyenv/versions/rasa2/lib/python3.8/site-packages/rasa/shared/core/domain.py", line 275, in merge_lists
    return sorted(list(set(list1 + list2)))
TypeError: unhashable type: 'dict'
```

**Comment by dgradwell-ams** [comment]
Hi I was wondering what release this bugfix was going to target? Our team is essentially blocked without it.

**Comment by melindaloubser1** [comment]
This will be fixed when 3.1 is released
